### PR TITLE
Add UI support for controlling the light app

### PR DESCRIPTION
**Pull request by andy31415**
Make the UI actually be able to toggle the light on/off and update brightness.

Tested by pairing the app with a controller and seeing the controller updating based on UI updates.

image

### CODE REVIEW

File: examples/common/imgui_ui/windows/light.cpp [hunk 1]
```diff
@@ -65,9 +65,18 @@ ImVec4 HueSaturationToColor(float hueDegrees, float saturationPercent)
 
 void Light::UpdateState()
 {
+    if (mTargetLightIsOn.HasValue()) {
+        OnOff::Attributes::OnOff::Set(mEndpointId, mTargetLightIsOn.Value());
```
Comment (bzbarsky-apple) [inline examples/common/imgui_ui/windows/light.cpp:69]: Should this be calling `OnOffServer::setOnOffValue`?  There's a bunch of logic that has to run when the state changes other than setting this attribute...

As in, should we be pretending to be an On or Off command here instead of just writing the attribute?

File: examples/common/imgui_ui/windows/light.cpp [hunk 2]
```diff
@@ -65,9 +65,18 @@ ImVec4 HueSaturationToColor(float hueDegrees, float saturationPercent)
...
+        mTargetLightIsOn.ClearValue();
+    }
     OnOff::Attributes::OnOff::Get(mEndpointId, &mLightIsOn);
 
     // Level Control
+    if (mTargetLevel.HasValue()) {
+        LevelControl::Attributes::CurrentLevel::Set(mEndpointId, mTargetLevel.Value());
```
Comment (bzbarsky-apple) [inline examples/common/imgui_ui/windows/light.cpp:76]: Again, how does this interact with in-flight level transitions?  How does it interact with the spec-required interactions between level and on/off state?

Please make sure @jmartinez-silabs reviews this.

File: examples/common/imgui_ui/windows/light.cpp (hunk 1 above)
Comment (andy31415) [reply examples/common/imgui_ui/windows/light.cpp:69]: That makes sense. Looking now.

The coupling is a bit harsh there though: lots of ember direct calls and the CommandHandler object passed in to the methods is very complex (probably mocking is not a great idea). Will look on how to update.

Comment (andy31415) [reply examples/common/imgui_ui/windows/light.cpp]: Updated both this and level control.

File: src/app/clusters/on-off-server/on-off-server.cpp [hunk 3]
```diff
@@ -613,15 +613,14 @@ bool OnOffServer::areStartUpOnOffServerAttributesNonVolatile(EndpointId endpoint
  */
 EmberEventControl * OnOffServer::getEventControl(EndpointId endpoint)
 {
-    uint16_t index            = emberAfFindClusterServerEndpointIndex(endpoint, OnOff::Id);
-    EmberEventControl * event = nullptr;
+    static EmberEventControl eventControls[EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT];
```
Comment (bzbarsky-apple) [inline src/app/clusters/on-off-server/on-off-server.cpp:616]: This incurs extra codesize to check for init on every call, no?  Please leave `eventControls` where it was.  If you think it's worth moving, that should be a separate PR.

Is the problem that the array sizing depends on endpoint_config and that's a problem for the header include?  If so, making this a file static is probably better than having a function static....

Comment (andy31415) [reply src/app/clusters/on-off-server/on-off-server.cpp:616]: The problem is indeed the header declaration depending on the endpoint count. I had started with a file static but saw that it is only used in a single function so I switched it to function static.

I am quite sure the storage requirement for all 3 cases is identical (except maybe the function static would be optimized out if the function is never called). This should not incur any extra storage on any of the function calls.

Comment (bzbarsky-apple) [reply src/app/clusters/on-off-server/on-off-server.cpp:616]: It's not about storage, it's about checking whether the static has already been initialized and initializing it if not....

Comment (andy31415) [reply src/app/clusters/on-off-server/on-off-server.cpp]: I would have expected some init code too, however generally since these are BSS zero-sized it probably wont happen:

image

However to not worry about it, switched this to a file static now.

File: examples/common/imgui_ui/windows/BUILD.gn [hunk 4]
```diff
@@ -51,5 +51,9 @@ static_library("light") {
     "${chip_root}/third_party/imgui",
   ]
 
+  # TODO: this is because on-off-server.h is generally only available
+  #       in the app and we do not want to directly bind to lighting-app
+  check_includes = false
```
Comment (jmartinez-silabs) [inline examples/common/imgui_ui/windows/BUILD.gn:56]: Just wondering if you have a plan/idea how to deal with that. 

We had a similar issues in a silabs common source set that required the data_model source set as deps. We end up setting a build argument that specify the data model source_set to be used per example but was wondering if there were other options

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/common/imgui_ui/windows/BUILD.gn (hunk 4 above)
Comment (andy31415) [reply examples/common/imgui_ui/windows/BUILD.gn:56]: Ideally I believe each server should become stand-alone and not depend on generated files, in which case `light_ui` would be able to depend on `light_server` without pulling in generated files.

However this is probably non-trivial to do today as many implementations require generated data information like ENDPOINT_COUNT. we may need to incrementally decouple things whenever possible.

Unfortunately this is not a plan - it looks like this level of decoupling involves major changes with the only gain being maintenance rather than feature support, so it will not be easy to get resources for such work.

File: src/app/clusters/on-off-server/on-off-server.cpp (hunk 3 above)
Comment (andy31415) [reply src/app/clusters/on-off-server/on-off-server.cpp]: Current UI lets the app instantiate it, so app can instantiate it on the endpoints it needs. It is indeed not based on codegen: one would need to change main to match generated code which is some duplication, however more generic.

It is assumed that the app knows exactly how many endpoints (and which ones) have light on it.

**Review by bzbarsky-apple — APPROVED** [review]
